examples/step-10/step-10.cc

```diff
@@ -54,7 +55,7 @@ namespace Step10
   // computed approximation to $\pi$ being in the range of the number of
   // digits which a double variable can hold, we rather declare the reference
   // value as a <code>long double</code> and give it a number of extra digits:
-  const long double pi = 3.141592653589793238462643L;
+  constexpr long double pi = 3.141592653589793238462643L;
```

Comment:
Didn't you want to get rid of `long double` here? What is the point of making it `constexpr`? The right-hand-side is a constant already.

Reply:
@masterleinad  

1. Yes, I shall get rid of the `long double` and update the results to close the other issue once this PR is merged. There were already too many changes in this PR, and this is not linked to any issue. These are general documentation fixes to step-10.

2. `constexpr` guarantees compile-time allocation/optimisation. This exact issue (using `constexpr` for the value of pi and why `constexpr` is better is answered [here](https://stackoverflow.com/questions/13346879/const-vs-constexpr-on-variables)

Reply:
Since we are not using it in any `constexpr` context here, there is essentially no difference at all and I don't see a good reason for changing it. 
If you are going to make it `double`, there is no reason to define the variable at all. You can just use `numbers::PI` instead.

Reply:
Ok. Let us just restore what we originally had. I will study `numbers::PI` later on and issue a PR later on.